controllers/copy.go

```diff
@@ -22,7 +22,23 @@ func (r *StatefulSetReconciler) copyPVC(ctx context.Context, src client.ObjectKe
 		return false, errors.New("unable to copy across namespaces")
 	}
 
-	job := newJob(src.Namespace, r.SyncContainerImage, src.Name, dst.Name)
+	var rbacObjs RbacObjects
+	if v := ctx.Value(RbacObjCtxKey); v != nil {
+		objs, ok := v.(RbacObjects)
+		if !ok {
+			return false, errors.New("Unexpected type for job RBAC config in context")
+		}
+		rbacObjs = objs
+	} else {
+		return false, errors.New("unable to extract job RBAC config from context")
+	}
```

Comment:
I probably would have just called `rbacObjName` again to get the `saname` instead of passing additional arguments through the context.

IMO passing mandatory argument through context is bad practice. But I've seen it before and some people would probably disagree, so I'm also OK with leaving it.